### PR TITLE
Cloud word stems

### PR DESCRIPTION
**Pull request by prashantiA**
- Updated association analysis to compare the roots of words instead of words directly
- Centralized comparison between keys in AssociationKey.java

### CODE REVIEW

File: capstone/src/main/java/com/google/sps/AssociationAnalysis.java [hunk 1]
```diff
@@ -3,20 +3,24 @@
 import java.util.ArrayList;
 import java.util.Collections;
 import java.util.HashMap;
+import opennlp.tools.stemmer.PorterStemmer;
```
Comment (nitinramadoss) [inline capstone/src/main/java/com/google/sps/AssociationAnalysis.java]: Have you made sure that you can use this module for commercial use?

File: capstone/src/main/java/com/google/sps/AssociationAnalysis.java [hunk 2]
```diff
@@ -3,20 +3,24 @@
...
 
 /** Calculates association scores from a list of sentiments per entity mention */
 public class AssociationAnalysis {
 
   HashMap<String, AssociationResult> res;
```
Comment (nitinramadoss) [inline capstone/src/main/java/com/google/sps/AssociationAnalysis.java]: You can declare these fields as private to the AssociationAnalysis class to maintain encapsulation.

File: capstone/src/main/java/com/google/sps/AssociationKey.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
+package com.google.sps;
+
+import opennlp.tools.stemmer.PorterStemmer;
+
+/** Generates lowercase no spaces word stems for keys in the AssociationResults hashmap */
```
Comment (nitinramadoss) [inline capstone/src/main/java/com/google/sps/AssociationKey.java:5]: Thanks for making this a separate class, makes it easier to understand :)

File: capstone/src/main/java/com/google/sps/AssociationAnalysis.java (hunk 1 above)
Comment (prashantiA) [reply capstone/src/main/java/com/google/sps/AssociationAnalysis.java]: Yeah I checked its licensed under the Apache License.

File: capstone/src/main/java/com/google/sps/AssociationAnalysis.java (hunk 2 above)
Comment (prashantiA) [reply capstone/src/main/java/com/google/sps/AssociationAnalysis.java]: Fixed this!

**Pull request merged**